ddtrace/llmobs/_llmobs.py

```diff
@@ -583,8 +584,88 @@ def create_dataset(cls, name: str, description: str, records: List[DatasetRecord
         ds = cls._instance._dne_client.dataset_create(name, description)
         for r in records:
             ds.append(r)
-        if len(records) > 0:
-            ds.push()
+        return ds
+
+    @classmethod
+    def create_dataset_from_csv(
+        cls,
+        csv_path: str,
+        dataset_name: str,
+        input_data_columns: List[str],
+        expected_output_columns: List[str],
+        metadata_columns: List[str] = [],
+        csv_delimiter: str = ",",
+        description="",
+    ) -> Dataset:
+        ds = cls._instance._dne_client.dataset_create(dataset_name, description)
+
+        # Store the original field size limit to restore it later
+        original_field_size_limit = csv.field_size_limit()
+
+        csv.field_size_limit(10 * 1024 * 1024)  # 10mb
+
+        try:
+            # First check if the file exists and is not empty before parsing
+            with open(csv_path, mode="r") as csvfile:
+                content = csvfile.readline().strip()
+                if not content:
+                    raise ValueError("CSV file appears to be empty or header is missing.")
+
+            with open(csv_path, mode="r") as csvfile:
```

Comment:
Instead of opening the file twice, can we just move the internal file pointer to the beginning of the file after checking the first line?
```suggestion
        csvfile.seek(0)
```